weblate_web/crm/views.py

```diff
@@ -336,3 +329,406 @@ def render_to_response(self, context, **response_kwargs):
+
+
+class IncomeView(CRMMixin, TemplateView):  # type: ignore[misc]
+    template_name = "crm/income.html"
+    permission = "invoices.view_income"
+    title = "Income Tracking"
+
+    # Chart configuration
+    CHART_WIDTH = 800
+    CHART_HEIGHT = 400
+    CHART_PADDING = 60
+    MIN_CHART_VALUE = Decimal(1)
+
+    # Category colors shared across all charts (keyed by category value)
+    CATEGORY_COLORS = {
+        InvoiceCategory.HOSTING: "#417690",
+        InvoiceCategory.SUPPORT: "#79aec8",
+        InvoiceCategory.DEVEL: "#5b80b2",
+        InvoiceCategory.DONATE: "#9fc5e8",
+    }
+
+    # Cached label-to-enum mapping (built once on first access)
+    _LABEL_TO_CATEGORY_CACHE: dict[str, InvoiceCategory] | None = None
+
+    @classmethod
+    def _get_label_to_category_map(cls) -> dict[str, InvoiceCategory]:
+        """Get or build cached label-to-enum mapping."""
+        if cls._LABEL_TO_CATEGORY_CACHE is None:
+            cls._LABEL_TO_CATEGORY_CACHE = {
+                category.label: category for category in InvoiceCategory
+            }
+        return cls._LABEL_TO_CATEGORY_CACHE
+
+    def _get_category_by_label(self, label: str) -> InvoiceCategory | None:
+        """Get category enum by its label."""
+        return self._get_label_to_category_map().get(label)
+
+    def get_year(self) -> int:
+        """Get the year from URL kwargs or default to current year."""
+        return self.kwargs.get("year", timezone.now().year)
+
+    def get_month(self) -> int | None:
+        """Get the month from URL kwargs if present."""
+        return self.kwargs.get("month")
+
+    def get_title(self) -> str:
+        year = self.get_year()
+        month = self.get_month()
+        if month:
+            return f"Income Tracking - {year}/{month:02d}"
+        return f"Income Tracking - {year}"
+
+    def generate_svg_pie_chart(self, data: dict[str, Decimal]) -> str:  # noqa: PLR0914
+        """Generate a simple SVG pie chart for category distribution with legend."""
+        if not data or sum(data.values()) == 0:
+            return ""
+
+        # Calculate required width based on legend text
+        # Longest category name is "Development / Consultations"
+        # Estimate: 420 (legend_x) + 20 (icon) + 300 (text) = 740, round to 750
+        width = 750
+        height = 400
+        radius = 120
+        center_x = 200
+        center_y = 200
+
+        total = sum(data.values())
+
+        # Count non-zero categories
+        non_zero_categories = [cat for cat, val in data.items() if val > 0]
+
+        svg_parts = [
+            f'<svg width="{width}" height="{height}" xmlns="http://www.w3.org/2000/svg" class="pie-chart">',
+        ]
+
+        # Special case: if only one category, draw a circle instead of a pie slice
+        if len(non_zero_categories) == 1:
+            category_label = non_zero_categories[0]
+            value = data[category_label]
+            category_enum = self._get_category_by_label(category_label)
```

Comment:
The generate_svg_pie_chart should rather get InvoiceCategory to avoid need to map it back to its value.